usr/share/rear/lib/bootloader-functions.sh

```diff
@@ -529,6 +529,58 @@ function get_root_disk_UUID {
     ( set -o pipefail ; mount | grep ' on / ' | awk '{print $1}' | xargs blkid -s UUID -o value || Error "Failed to get root disk UUID" )
 }
 
+# Detect whether actually GRUB 2 is installed and that test is to
+# check if grub-probe or grub2-probe is installed because
+# grub-probe or grub2-probe is only installed in case of GRUB 2.
+# Needed because one can't tell the GRUB version by looking at the MBR
+# (both GRUB 2 and GRUB Legacy have the string "GRUB" in their MBR).
+function is_grub2_installed () {
+    if type -p grub-probe >&2 || type -p grub2-probe >&2 ; then
+        LogPrint "GRUB 2 is installed (grub-probe or grub2-probe exist)."
```

Comment:
I think DebugPrint would be better here
to avoid that this message is shown to the user
every time when is_grub2_installed is called.
Perhaps even only Log is sufficient here.
When needed user messages can be shown
at the place where is_grub2_installed is called.

Reply:
@jsmeix I can do that. I chose this, as the original code ( https://github.com/rear/rear/pull/3145/files#diff-078a2efa46c4e9db15409dc78a15393011d85d51b0dba2323d590e06f3db36b3L47 ) had `LogPrint` as well, so I wanted to preserve the information. It is true that it may be a bit too verbose, so I will do as you suggest.

Reply:
I think I will use `Log` here, `DebugPrint` would make the message disappear even from the log if not DEBUG, and I prefer to have more verbose logs. Or, I could introduce `LogDebugPrint` that would log always and print only if DEBUG.

Reply:
At the time when I implemented that in June 2016 via
https://github.com/rear/rear/commit/6eb37d9f32a862fba168b74b554fe5032bd9c17f
there was not yet a DebugPrint function so I used LogPrint.

Over the time such LogPrint calls became more and more
so ReaR's verbose mode became more and more oververbose.

I implemented the DebugPrint function in July 2017 via
https://github.com/rear/rear/commit/2a03091340db764b3f3f7598b1dfd3bdeaf6bcd0

There are many older LogPrint calls where it is
questionable if they are helpful in verbose mode
and where DebugPrint is probably better.

Reply:
I don't know if having a message like
```
GRUB 2 is installed (grub-probe or grub2-probe exist).
```
only in the log file (but not on the user's terminal)
could be useful or helpful in production environments?

Reply:
Offhandedly I think instead of an additional LogDebugPrint
it would be better to change the current DebugPrint
to log always and print only if DEBUG.

In particular when you prefer to have more verbose logs
in any case.

I also want verbose logs and I call ReaR basically
always in debug mode to get verbose logs.

But when there are more verbose logs in any case
some users may not like that and probabyl righfully
complain why the logs are so verbose in any case
when there is the verbose mode and the debug mode
for those who want more verbose logs.

I think the messaging/logging area is complicated
and it should be better discussed separatedly.

For this particular case here feel free to use
what you think is best here.

Reply:
> Personally I don't use Log() because for my personal needs
> having something only in the log file is not helpful.
> I never look at the log file unless I need to investigate
> something and then I want debug messages.

the idea is that especially during recovery, one wants to have information in the log by default and not only when asked, because if a user encounters a problem but the recovered machine basically works, they may dislike to have to reproduce the issue by recovering again with debug mode, so the log from the run with default settings will be the only thing that you will have.

Reply:
done in [4be6d2f](https://github.com/rear/rear/pull/3145/commits/4be6d2f9853f65d30f3c818c45fdbe0d05598207)